wgpu-core/src/command/compute.rs

```diff
@@ -870,12 +888,132 @@ fn dispatch_indirect(
             MemoryInitKind::NeedsInitializedMemory,
         ));
 
-    state.flush_states(Some(buffer.tracker_index()))?;
+    #[cfg(feature = "indirect-validation")]
+    {
+        let params = state.device.indirect_validation.as_ref().unwrap().params(
+            &state.device.limits,
+            offset,
+            buffer.size,
+        );
 
-    let buf_raw = buffer.try_raw(&state.snatch_guard)?;
-    unsafe {
-        state.raw_encoder.dispatch_indirect(buf_raw, offset);
+        unsafe {
+            state.raw_encoder.set_compute_pipeline(params.pipeline);
+        }
+
+        unsafe {
+            state.raw_encoder.set_push_constants(
+                params.pipeline_layout,
+                wgt::ShaderStages::COMPUTE,
+                0,
+                &[params.offset_remainder as u32 / 4],
+            );
+        }
+
+        unsafe {
+            state.raw_encoder.set_bind_group(
+                params.pipeline_layout,
+                0,
+                Some(params.dst_bind_group),
+                &[],
+            );
+        }
+        unsafe {
+            state.raw_encoder.set_bind_group(
+                params.pipeline_layout,
+                1,
+                Some(
+                    buffer
+                        .raw_indirect_validation_bind_group
+                        .get(&state.snatch_guard)
+                        .unwrap()
+                        .as_ref(),
+                ),
+                &[params.aligned_offset as u32],
+            );
+        }
+
+        let src_transition = state
+            .intermediate_trackers
+            .buffers
+            .set_single(&buffer, hal::BufferUses::STORAGE_READ);
+        let src_barrier =
+            src_transition.map(|transition| transition.into_hal(&buffer, &state.snatch_guard));
+        unsafe {
+            state.raw_encoder.transition_buffers(src_barrier.as_slice());
+        }
+
+        unsafe {
+            state.raw_encoder.transition_buffers(&[hal::BufferBarrier {
+                buffer: params.dst_buffer,
+                usage: hal::BufferUses::INDIRECT..hal::BufferUses::STORAGE_READ_WRITE,
```

Comment:
issue: This range seems invalid; `INDIRECT` is an element that is _after_ `STORAGE_READ_WRITE`. Was the intent to provide an empty range? If so, there's probably clearer way to do it. If not, theeeen oops!

Reply:
This looks an inverted version of a similar call to `transition_buffers` that uses 	`usage: hal::BufferUses::STORAGE_READ_WRITE..hal::BufferUses::INDIRECT`.

Reply:
@cwfitzgerald explained to me that we're subverting the `Range` type to be a data structure of a "from" and "to" pair, rather than a range of values. That explains my misunderstanding!

I…think we should split `usage` into `*from` and `*to` pairs, because I would expect a similar confusion from anybody else with a background in idiomatic Rust. 😅 That need not block here, but it'd be a nice papercut to remove for later.